EOSBasic/Plugins/UEOS/Source/UEOS/Private/Presence/Presence.cpp

```diff
@@ -0,0 +1,147 @@
+// Copyright (C) Gaslight Games Ltd, 2019-2020
+
+#pragma once
+
+// UEOS Includes
+#include "Presence/Presence.h"
+#include "UEOSManager.h"
+#include "UserInfo/UserInfo.h"
+
+#include "UEOSModule.h"
+
+//TODO - Class is in the works - will be used with friend interface - Mikhail 
+
+UEOSPresence::UEOSPresence()
+{
+
+}
+
+EPresenceStatus UEOSPresence::GetPresenceStatus(const FEpicAccountId& InFriendInfo)
+{
+	EOS_HPresence PresenceHandle = EOS_Platform_GetPresenceInterface(UEOSManager::GetPlatformHandle());
+	
+	EOS_Presence_CopyPresenceOptions CopyOptions;
+	CopyOptions.ApiVersion = EOS_PRESENCE_COPYPRESENCE_API_LATEST;
+	CopyOptions.LocalUserId = UEOSManager::GetAuthentication()->GetEpicAccountId();
+
+	FEpicAccountId AccountId = FEpicAccountId();
+	CopyOptions.TargetUserId = AccountId.FromString(InFriendInfo.ToString());
+
+	EOS_Presence_Info* PresenceInfo = nullptr;
+	EOS_EResult ResultCode = EOS_Presence_CopyPresence(PresenceHandle, &CopyOptions, &PresenceInfo);
+
+	if (ResultCode != EOS_EResult::EOS_Success)
+	{
+		UE_LOG(UEOSLog, Warning, TEXT("[EOS SDK | Plugin] Error when getting presence status: %s"), *UEOSCommon::EOSResultToString(ResultCode));
+	}
+	
+	EPresenceStatus PresenceStatus = (EPresenceStatus)PresenceInfo->Status;
+
+	EOS_Presence_Info_Release(PresenceInfo);
+
+	return PresenceStatus;
+}
+
+void UEOSPresence::AddFriendPresence(const FEpicAccountId& InFriendInfo)
+{
+	EOS_HPresence PresenceHandle = EOS_Platform_GetPresenceInterface(UEOSManager::GetPlatformHandle());
+	FEpicAccountId AccountId = FEpicAccountId();
+	EOS_Presence_QueryPresenceOptions Options;
+	Options.ApiVersion = EOS_PRESENCE_QUERYPRESENCE_API_LATEST;
+	Options.LocalUserId = UEOSManager::GetAuthentication()->GetEpicAccountId();
+	Options.TargetUserId = AccountId.FromString(InFriendInfo.ToString());
+	EOS_Presence_QueryPresence(PresenceHandle, &Options, NULL, QueryPresenceCompleteCallback);
+}
+
+void UEOSPresence::QueryPresenceCompleteCallback(const EOS_Presence_QueryPresenceCallbackInfo* Data)
+{
+	check(Data != nullptr);
```

Comment:
Consider switching these to nullptr ifs, so we can catch without crashing the editor. And/or consider using checkf or other, so can print some information.